### PR TITLE
Fix library name on macOS

### LINKED ISSUE #80: Versioning of macOS library
In the conda-forge distribution, we are seeing errors for programs that were built against antic 0.2.4 when installing 0.2.5. 

```
cling::DynamicLibraryManager::loadLibrary() [/Users/runner/miniforge3/conda-bld/flatsurf_1659591997916/[…]/lib/libexactreal.6.dylib, 9):
Library not loaded: @rpath/libantic-0.2.4.dylib
```

I don't really understand how library naming works on macOS but could it be that this library should be called `libantic.0.dylib` instead of `libantic-0.2.4.dylib`?

The relevant part of the configure script is [here](https://github.com/wbhart/antic/blob/trunk/configure#L384-L386):

```sh
ANTIC_LIBNAME="libantic.dylib"
ANTIC_LIB="libantic-$ANTIC_MAJOR.$ANTIC_MINOR.$ANTIC_PATCH.dylib"
EXTRA_SHARED_FLAGS="-install_name $PREFIX/lib/$ANTIC_LIB -compatibility_version $ANTIC_MAJOR.$ANTIC_MINOR -current_version $ANTIC_MAJOR.$ANTIC_MINOR.$ANTIC_PATCH";;
```

According to [Apple Docs](https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/DynamicLibraries/100-Articles/CreatingDynamicLibraries.html), we should probably change this. It says:

> <p class="codesample clear"><strong class="caption_number">Listing 3</strong>  Generating version 1.0 of the Ratings dynamic library</p><div class="codesample clear">
> [Ratings/1.0]% make dylib
> `clang -dynamiclib -std=gnu99 Ratings.c -current_version 1.0 -compatibility_version 1.0 -fvisibility=hidden -o libRatings.A.dylib`
> </div><p>This list indicates where the major version, minor version, and compatibility version are specified:</p><ul class="ul"><li class="li"><p>The major version number is specified in the library’s filename as “A” in <code>-o libRatings.A.dylib</code>.</p></li><li class="li"><p>The minor version number is specified in <code>-current_version 1.0</code>.</p></li><li class="li"><p>The compatibility version number is specified in <code>-compatibility_version 1.0</code>.</p></li></ul>

### PR DESCRIPTION
**Pull request by saraedum**
According to the docs at https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/DynamicLibraries/100-Articles/CreatingDynamicLibraries.html

> The major version number is specified in the library's filename as "A" in -o libratings.A.dylib.

Fixes #80.

---

But I guess the above documentation assumes that the major number enodes ABI backwards compatibility.

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -382,17 +382,17 @@ if [ -z "$ANTIC_LIB" ]; then
    case "$OS" in
       Darwin)
          ANTIC_LIBNAME="libantic.dylib"
-	 ANTIC_LIB="libantic-$ANTIC_MAJOR.$ANTIC_MINOR.$ANTIC_PATCH.dylib"
+         ANTIC_LIB="libantic.$ANTIC_MAJOR.dylib"
          EXTRA_SHARED_FLAGS="-install_name $PREFIX/lib/$ANTIC_LIB -compatibility_version $ANTIC_MAJOR.$ANTIC_MINOR -current_version $ANTIC_MAJOR.$ANTIC_MINOR.$ANTIC_PATCH";;
```
Comment (isuruf) [inline configure:380]:
```suggestion
         EXTRA_SHARED_FLAGS="-install_name $PREFIX/lib/$ANTIC_LIB -compatibility_version $ANTIC_MAJOR -current_version $ANTIC_MAJOR.$ANTIC_MINOR.$ANTIC_PATCH";;
```

File: configure [hunk 2]
```diff
@@ -382,17 +382,17 @@ if [ -z "$ANTIC_LIB" ]; then
    case "$OS" in
       Darwin)
          ANTIC_LIBNAME="libantic.dylib"
-	 ANTIC_LIB="libantic-$ANTIC_MAJOR.$ANTIC_MINOR.$ANTIC_PATCH.dylib"
-         EXTRA_SHARED_FLAGS="-install_name $PREFIX/lib/$ANTIC_LIB -compatibility_version $ANTIC_MAJOR.$ANTIC_MINOR -current_version $ANTIC_MAJOR.$ANTIC_MINOR.$ANTIC_PATCH";;
+         ANTIC_LIB="libantic.$ANTIC_MAJOR.dylib"
+         EXTRA_SHARED_FLAGS="-install_name $PREFIX/lib/$ANTIC_LIB -compatibility_version $ANTIC_MAJOR -current_version $ANTIC_MAJOR.$ANTIC_MINOR.$ANTIC_PATCH";;
```
Comment (saraedum) [inline configure]:
```suggestion
         EXTRA_SHARED_FLAGS="-install_name $PREFIX/lib/$ANTIC_LIB -compatibility_version $ANTIC_MAJOR.$ANTIC_MINOR -current_version $ANTIC_MAJOR.$ANTIC_MINOR.$ANTIC_PATCH";;
```

Actually, I don't think that antic is following semantic versioning @isuruf so let's keep it at x.x. That's also what our antic-feedstock over at conda-forge does.

Comment (isuruf) [reply configure]: Then the SONAME in Linux should reflect that too. Currently the SONAME uses only `MAJOR_VERSION`

Comment (saraedum) [reply configure]: @isuruf I guess you're right. So my (limited) understanding is that we should compile into
`libantic.so.major.minor.patch` and then symlink `libantic.so` and `libantic.so.major.minor`.

That's the situation on Linux.

What's the correct thing to do on macOS and Windows? Essentially the same? I.e.,

Windows; Compile into `libantic-major.minor.patch.dll` and "symlink" this to `libantic.dll` and `libantic-major.minor.dll`? (It seems to me that Windows does not do versioning in .dll names usually; at least I could not find much about it online.)

macOS: Compile into `libantic.major.minor.patch.dylib` and symlink there from `libantic.dylib` and `libantic.major.minor.dylib`?

Comment (saraedum) [reply configure]: Or is the only important thing that we set the `-soname` flags correctly and we should not worry too much about the naming of the files?

Comment (isuruf) [reply configure]: > So my (limited) understanding is that we should compile into
libantic.so.major.minor.patch and then symlink libantic.so and libantic.so.major.minor.

Yes. SONAME should be then `libantic.so.major.minor`. Similar for macOS.

On windows, directly compile into `libantic-major.minor.dll` and no symlinks.

**Review by isuruf — APPROVED** [review]

**Comment by saraedum** [comment]
@fredrik-johansson could this be merged?

**Comment by fredrik-johansson** [comment]
Certainly.

**Pull request merged**